docs/source/ros_api.rst

```diff
@@ -0,0 +1,117 @@
+.. Copyright 2016 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+=======
+ROS API
+=======
+
+The following ROS API is provided by `cartographer_node`_.
+
+Command-line Flags
+==================
+
+.. TODO(damonkoler): Use an options list if it can be made to render nicely.
+
+\-\-configuration_directory
+  First directory in which configuration files are searched, "second is always
+  the Cartographer installation to allow including files from there.
+
+\-\-configuration_basename
+  Basename, i.e. not containing any directory prefix, of the configuration file.
+
+Subscribed Topics
+=================
+
+The following range data topics are mutually exclusive.
+
+scan (`sensor_msgs/LaserScan`_)
+  Currently only supported in 2D. If *use_horizontal_laser* is enabled in the
+  :doc:`configuration`, this topic will be used as input for SLAM.
+
+echoes (`sensor_msgs/MultiEchoLaserScan`_)
+  Currently only supported in 2D. If *use_horizontal_multi_echo_laser* is
+  enabled in the :doc:`configuration`, this topic will be used as input for
+  SLAM. Only the first echo is used.
+
+points2 (`sensor_msgs/PointCloud2`_)
+  Currently only supported in 3D. If *num_lasers_3d* is set to 1 in the
+  :doc:`configuration`, this topic will be used as input for SLAM. If
+  *num_lasers_3d* is greater than 1, multiple numbered points2 topics (i.e.
+  points2_1, points2_2, points2_3, ...  up to and including *num_lasers_3d*)
+  will be used as inputs for SLAM.
+
+The following additional sensor data topics are optional.
+
+imu (`sensor_msgs/Imu`_)
+  Supported in 2D and 3D. This topic will be used as input for SLAM.
+
+odom (`nav_msgs/Odometry`_)
+  Supported in 2D and 3D. If *use_odometry_data* is enabled in the
+  :doc:`configuration`, this topic will be used as input for SLAM.
+
+Published Topics
+================
+
+map (`nav_msgs/OccupancyGrid`_)
+  Currently only supported in 2D. If subscribed to, a background thread will
+  continuously compute and publish the map. Depending on the size of the map, it
+  can take a few seconds between updates.
+
+scan_matched_points2 (`sensor_msgs/PointCloud2`_)
+  Point cloud as it was used for the purpose of scan-to-submap matching.
```

Comment:
This might contain less points than the laser topics, since filtering takes place before scan matching, i.e. these are the points the SLAM algorithm sees.